flocker/common/plugin.py

```diff
@@ -0,0 +1,135 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Tools for loading third-party plugins.
+"""
+
+from characteristic import attributes
+from pyrsistent import PClass, field, PVector, pvector
+from twisted.python.reflect import namedAny
+
+
+@attributes(["plugin_name"])
+class PluginNotFound(Exception):
+    """
+    A plugin with the given name was not found.
+
+    :attr str plugin_name: Name of the plugin looked for.
+    """
+    def __str__(self):
+        return (
+            "'{!s}' is neither a built-in plugin nor a 3rd party "
+            "module.".format(self.plugin_name)
+        )
+
+
+class InvalidPlugin(Exception, object):
```

Comment:
I added that when trying to debug https://github.com/hynek/characteristic/issues/37. It isn't needed.